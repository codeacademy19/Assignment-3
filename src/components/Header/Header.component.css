/* 
header{
    background-color: #555;
    text-align: center;
    position: relative;
    z-index: 1;
    width:100 vw;
    
}
header:before {
        border-top: 3px solid white;
        content:"";
        margin: 0 auto; 
        position: absolute; 
        top: 50%; left: 0; right: 0; bottom: 0;
        width: 100vw;
        z-index: -1;
        right: 5%;
    }

 .headerSpan { 
        /* display: block;
        background: #fff; 
        padding: 10px ;  */
/* } */ 

/* header:before {
    content: "";
    display: block;
    border-top: solid 3px white;
    width: 100%;
    height: 1px;
    position: absolute;
    top: 5%;
    z-index: 1;
  }
  
  .spanHeader {
      display: inline-block;
    background: #fff;
    padding: 0 20px;
    position: relative;
    z-index: -1;
  } */
header{
    display: flex;
    justify-content: space-evenly;
}
.hr1{
    align-items: center;
    align-self: center;
    width:100%;
}
.headerSpan{
    width:50%;
}
.hr2{
    align-self: center;
    width:100%;
}
.BS{ 
    font-size: 3rem;
    display:inline-flex;
    padding: 10px;
    border: 2px solid rgb(246, 243, 243);
    align-items: center;
    justify-content: center;
    background-color: #555;
    color: white;
    min-width: 1em;
    border-radius: 100%;
    vertical-align: middle;
}
.description{
    font-size: 2em;
    color: white;
    font-style: italic;
    font-family: sans-serif;
} 
@media screen and (max-width: 500px) {
    .description{
        display:none;
    }
    .hr1, .hr2{
        display: none;
    }
}
