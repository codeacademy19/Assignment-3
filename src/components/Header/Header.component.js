import React, { Component } from 'react';
import './Header.component.css';

class Header extends Component {
  render() {
    return (
      <header>
        <div className="hr1"><hr /></div>
        <div className="headerSpan">
          <div className="BS">BS</div>
          <div className="description">The Book Shelf</div>
        </div>

        <div className="hr2"><hr /></div>
      </header>
    );
  }
}

export default Header;
