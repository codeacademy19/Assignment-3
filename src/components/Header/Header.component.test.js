import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header.component';


describe('Header', () => {
  it('should render header', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
