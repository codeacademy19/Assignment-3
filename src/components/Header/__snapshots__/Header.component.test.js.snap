// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render header 1`] = `
<header>
  <div
    className="hr1"
  >
    <hr />
  </div>
  <div
    className="headerSpan"
  >
    <div
      className="BS"
    >
      BS
    </div>
    <div
      className="description"
    >
      The Book Shelf
    </div>
  </div>
  <div
    className="hr2"
  >
    <hr />
  </div>
</header>
`;
